h3napi.c

```diff
@@ -42,24 +47,28 @@
   napi_value V;\
   if (napi_create_array_with_length(env, L, &V) != napi_ok) {\
     napi_throw_error(env, "ENOSPC", "Could not create fixed length array");\
+    return NULL;\
   }
 
 #define napiVarArray(V) \
   napi_value V;\
   if (napi_create_array(env, &V) != napi_ok) {\
     napi_throw_error(env, "ENOSPC", "Could not create variable array");\
+    return NULL;\
```

Comment:
It looks like this would cause a memory leak, for example around line 256 (`napiFn(kRing)`) the `calloc`'d array leaks. If an error is thrown does the entire process get torn down?

Reply:
Good point. Let me think about how I can handle this appropriately.

Reply:
So I can have the template end with another if statement and end the line with a block of `free`s that are only run when exiting early.